ports/usockets/CMakeLists.txt

```
@@ -1,68 +1,158 @@
-cmake_minimum_required(VERSION 3.13)
-project(uSockets C CXX)
+# Unofficial uSockets CMakeLists.txt from Makefile of https://github.com/uNetworking/uSockets/blob/v0.8.8/Makefile
+cmake_minimum_required(VERSION 3.25.1)
+project(uSockets VERSION ${VERSION} LANGUAGES C CXX)
 
-option(INSTALL_HEADERS "Install header files" ON)
+# options
 
-if (CMAKE_USE_OPENSSL)
-    find_package(OpenSSL REQUIRED)
-    set(USE_OPENSSL "-DUSE_OPENSSL -DLIBUS_USE_OPENSSL")
-    #set(OPENSSL_LIB "OpenSSL::SSL OpenSSL::Crypto")
-    list(APPEND CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-else()
-    set(NOT_USE_OPENSSL "-DLIBUS_NO_SSL")
-endif()
+option(WITH_BORINGSSL   "enables BoringSSL support (preferred over OpenSSL)"                OFF)
+option(WITH_OPENSSL     "enables OpenSSL 1.1+ support"                                      OFF)
+option(WITH_WOLFSSL     "enables WolfSSL 4.2.0 support (mutually exclusive with OpenSSL)"   OFF)
+option(WITH_LIBUV       "builds with libuv as event-loop"                                   OFF)
+
+# global variables
+
+set(uSockets_installable_libs)
+set(uSockets_find_dependency_calls)
+
+# target usockets
+
+add_library(uSockets)
+list(APPEND uSockets_installable_libs uSockets)
+foreach(dir IN ITEMS "src" "src/eventing" "src/crypto" "src/io_uring")
+    aux_source_directory(${dir} uSockets_sources)
+endforeach()
+list(FILTER uSockets_sources INCLUDE REGEX ".+\\.c$")
+target_sources(uSockets PRIVATE ${uSockets_sources})
+if(WIN32)
+    target_link_libraries(uSockets PRIVATE ws2_32)
+endif(WIN32)
+
+# target coupling options
+
+add_library(us_target_coupling INTERFACE)
+list(APPEND uSockets_installable_libs us_target_coupling)
+# global include
+target_include_directories(us_target_coupling INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>")
 
-find_package(libuv CONFIG REQUIRED)
-if (TARGET libuv::uv)
-    set(LIBUV_LIBRARY libuv::uv)
+# global std
+target_compile_features(us_target_coupling INTERFACE "c_std_11")
+if(WITH_OPENSSL OR WITH_BORINGSSL)
+    target_compile_features(us_target_coupling INTERFACE "cxx_std_17")
 else()
-    set(LIBUV_LIBRARY libuv::uv_a)
+    target_compile_features(us_target_coupling INTERFACE "cxx_std_14")
 endif()
 
-file(GLOB SOURCES src/*.c src/eventing/*.c)
+# ssl
 
-set(USOCKETS_EXT_INCLUDE_DIR )
-set(USOCKETS_EXT_LIBS )
+if(WITH_BORINGSSL)
+    target_compile_definitions(us_target_coupling INTERFACE "-DLIBUS_USE_BORINGSSL")
 
-if (CMAKE_USE_OPENSSL)
-    # It requires C++17 or later, see https://github.com/uNetworking/uSockets/blob/0ebdde0601cc82349fc11a7c4bbb6dc5c9f28f42/Makefile#L55
-    set(CMAKE_CXX_STANDARD 17)
     find_package(OpenSSL REQUIRED)
-    file(GLOB SSL_SOURCES src/crypto/*.c*)
-    list(APPEND SOURCES ${SSL_SOURCES})
-    list(APPEND USOCKETS_EXT_LIBS OpenSSL::SSL OpenSSL::Crypto)
-endif()
+    list(APPEND uSockets_find_dependency_calls "OpenSSL REQUIRED")
```

```suggestion
    list(APPEND uSockets_find_dependency_calls "OpenSSL")
```

find_dependency should never have REQUIRED. Ditto all other `list(APPEND uSockets_find_dependency_calls`s

---

Done. Removed these.